### DISCUSSION TITLE
Transfer data over USB

### DISCUSSION THREAD

**Discussion by Colpocorto** [ZuluSCSI/ZuluSCSI-firmware#130] [Ideas]
It would be great if the contents of the SD Card could be accessed via USB (as the SCSI2SD v6 does).

**Comment by aperezbios** [comment]
Given that it would be limited to USB 1.1 speeds, under one megabyte per second, would that still be useful to you, @Colpocorto ? Which hardware version of ZuluSCSI do you have? You did not specify.

**Reply by Colpocorto** [reply]
Hello, thank you for your response.

I have a RP2040 model. The thing is that I want to install the Zulu internally in a sampler, but I need very frequent data exchange between the sampler and the computer.

USB1.1 speed would be a PITA but still better than nothing.

**Reply by PetteriAimonen** [reply]
One option worth considering is SD card extender cables, such as [this](https://www.adafruit.com/product/3688).
That would allow installing the SD card slot so that it is easy to access.

**Reply by Colpocorto** [reply]
> One option worth considering is SD card extender cables, such as [this](https://www.adafruit.com/product/3688). That would allow installing the SD card slot so that it is easy to access.

Excellent! I didn't know such device exists. It might be a nice workaround.

In any case, since I usually transfer tons of data from/to the SD card, it would be great to have a remote way to update data. Direct USB transfer (ala SCSI2SD v6) is great. A network connection (e.g. PiSCSI) would be perfect.

**Reply by EntropyCoM** [reply]
Hi There,
then what use is the USB-Port then, anyway?
I do not get it... 

Kind regards,
EntropyCoM

**Reply by aperezbios** [reply]
@EntropyCoM. The USB port is both a mechanism for firmware recovery, as well as an optional way to power the board itself. With the ZuluSCSI RP2040, the USB port can also currently be used to as a virtual serial port, to see the ZuluSCSI log file output in real-time, as it exposes itself as a USB virtual COM port. Additionally, the USB port can be used for other purposes in the future, such as things we haven't even thought of yet.

**Comment by aperezbios** [comment]
@EntropyCoM On the ZuluSCSI RP2040, the USB port can also be used to view the log output in real-time. The ZuluSCSI RP2040 exposes itself as a USB virtual UART/COM device.

**Comment by aperezbios** [comment]
@Colpocorto which specific sampler are you using your ZuluSCSI with?

**Reply by Colpocorto** [reply]
I don’t have a Zulu interface anymore. I sold it because of its limitations. I had it installed on a E-mu E4XT Ultra sampler.

**Reply by aperezbios** [reply]
@Colpocorto Which limitations were of primary concern to you? It would be helpful to understand more.

**Reply by Colpocorto** [reply]
> @Colpocorto Which limitations were of primary concern to you? It would be helpful to understand more.

Well, I opened this thread for a reason. Not having a way to directly transfer data between the main PC and the Zulu-equipped device is absolutely a no go for me.
